clusterloader2/pkg/util/cluster.go

```diff
@@ -88,11 +104,15 @@ func isNodeSchedulable(node *corev1.Node) bool {
 	return !node.Spec.Unschedulable && nodeReady && networkReady
 }
 
-// Tests whether node doesn't have any taint with "NoSchedule" or "NoExecute" effect.
+// Tests whether node doesn't have any built-in taint with "NoSchedule" or "NoExecute" effect.
 func isNodeUntainted(node *corev1.Node) bool {
-	for i := range node.Spec.Taints {
-		if node.Spec.Taints[i].Effect == corev1.TaintEffectNoSchedule || node.Spec.Taints[i].Effect == corev1.TaintEffectNoExecute {
-			return false
+	for _, nodeTaint := range node.Spec.Taints {
+		effect := nodeTaint.Effect
+		key := nodeTaint.Key
+		for _, evictingTaintKey := range builtInTaintsKeys {
+			if key == evictingTaintKey && (effect == corev1.TaintEffectNoSchedule || effect == corev1.TaintEffectNoExecute) {
```

Comment:
effect check could be done outside of the loop

Reply:
Done.